### PR TITLE
Bump reference counts to blocks of newly loaded file earlier

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #5862

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Review by crsib — DISMISSED** [review]
That's really elegant 😻

File: libraries/lib-wave-track/WaveTrack.cpp [hunk 1]
```diff
@@ -2359,17 +2359,12 @@ bool WaveTrack::FormatConsistencyCheck() const
       });
 }
 
-bool WaveTrack::InsertClip(WaveClipHolder clip)
+void WaveTrack::InsertClip(WaveClipHolder clip)
 {
-   if(!clip->GetIsPlaceholder() && clip->IsEmpty())
```
Comment (vsverchinsky) [inline libraries/lib-wave-track/WaveTrack.cpp:2364]: I think that the explanation of this line presense is that we've found out that zero clips could have been created during editing the project.

**Comment by Paul-Licameli** [comment]
Please review it again.  The second commit is very different now and makes more conservative changes.

**Review by vsverchinsky — APPROVED** [review]

**Comment by Paul-Licameli** [comment]
Fix the corrupting of projects that were saved with empty clips with smart data

**Pull request merged**